### PR TITLE
Document S3 internal memory properties.

### PR DESCRIPTION
**Pull request by bguedas**

### CODE REVIEW

File: ApplicationDeveloperGuide/applicationOptions.rst [hunk 1]
```diff
@@ -107,8 +107,14 @@ In order to export options defined in a ``.launch`` file to a properties file, p
 3. Browse the desired output ``.properties`` file,
 4. Press the :guilabel:`Finish` button.
 
+.. warning::
+   The Simulator uses some system properties to configure internal memory limits.
```
Comment (frivieremicroej) [inline ApplicationDeveloperGuide/applicationOptions.rst:111]: If these options cannot be set as application options, I suggest to introduce and move to a dedicated Simulation section in the ADG.

File: ApplicationDeveloperGuide/sectionSimulatorOptions.rst [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+^^^^^^^^^^^^^^^^^^^^^^
+
+
+*Option Name*: ``S3.JavaMemory.HeapSize``
+
+*Default value*: ``4096``
+
+*Description*:
+
+Heap Size allocated by the Simulator. It is used to allocate object on Java Heap,
```
Comment (frivieremicroej) [inline ApplicationDeveloperGuide/sectionSimulatorOptions.rst:32]: You need to increase this option if Java heap + immortal heap application options > 4MB, am I right  ?

File: ApplicationDeveloperGuide/sectionSimulatorOptions.rst [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+but also immutable objectss. It thus must be at least equal to the sum of option_java_heap_
```
Comment (frivieremicroej) [inline ApplicationDeveloperGuide/sectionSimulatorOptions.rst:33]:
```suggestion
but also immutable objects. It thus must be at least equal to the sum of option_java_heap_ 
```

File: ApplicationDeveloperGuide/sectionSimulatorOptions.rst [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+^^^^^^^^^^^^^^^^^^^^^^^^^
+
+
+*Option Name*: ``S3.JavaMemory.SystemCharsSize``
+
+*Default value*: ``100``
+
+*Launchscripts value*: ``1024``
+
+*Recomended value*: ``3000``
```
Comment (frivieremicroej) [inline ApplicationDeveloperGuide/sectionSimulatorOptions.rst:47]:
```suggestion
*Recommended value*: ``3000``
```

File: ApplicationDeveloperGuide/sectionSimulatorOptions.rst [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+
+*Description*:
+
+Memory allocated to system strings. Value is in kilobytes.
```
Comment (frivieremicroej) [inline ApplicationDeveloperGuide/sectionSimulatorOptions.rst:51]: "system strings" can you elaborate in which "application" case we will reach this limit (intern strings, application class names ?)

File: ApplicationDeveloperGuide/sectionSimulatorOptions.rst [hunk 6]
```diff
@@ -0,0 +1,151 @@
...
+
+*Option Name*: ``S3.JavaMemory.ApplicationCharsSize``
+
+*Default value*: ``250``
+
+*Launchscripts value*: ``4096``
+
+*Description*:
+
+Memory allocated to application strings. Value is in kilobytes.
```
Comment (frivieremicroej) [inline ApplicationDeveloperGuide/sectionSimulatorOptions.rst:66]: application strings / system strings: not clear (see comment above)

File: ApplicationDeveloperGuide/sectionSimulatorOptions.rst [hunk 7]
```diff
@@ -0,0 +1,151 @@
...
+*Description*:
+
+Memory space allocated for methods. Value is in kilobytes.
+
+
+Option: Thread Stack Size
+^^^^^^^^^^^^^^^^^^^^^^^^^
+
+
+*Option Name*: ``S3.JavaMemory.MethodsSize``
```
Comment (frivieremicroej) [inline ApplicationDeveloperGuide/sectionSimulatorOptions.rst:106]: ? same option name than before ?

File: ApplicationDeveloperGuide/sectionSimulatorOptions.rst [hunk 8]
```diff
@@ -0,0 +1,151 @@
...
+
+*Default value*: ``300``
+
+*Description*:
+
+Memory space allocated for each thread stack. Value is in kilobytes.
```
Comment (frivieremicroej) [inline ApplicationDeveloperGuide/sectionSimulatorOptions.rst:112]: do we already reached this option (compared to embedded constraints?)

File: ApplicationDeveloperGuide/sectionSimulatorOptions.rst [hunk 9]
```diff
@@ -0,0 +1,151 @@
...
+
+
+
+Group: Icetea Heap Size
+-----------------------
+
+Option: Icetea Heap Start
```
Comment (frivieremicroej) [inline ApplicationDeveloperGuide/sectionSimulatorOptions.rst:119]: I guess this value must not be changed and should be hidden

File: ApplicationDeveloperGuide/sectionSimulatorOptions.rst [hunk 10]
```diff
@@ -0,0 +1,151 @@
...
+
+*Option Name*: ``IceteaRuntimeSupport.S3.HeapEnd``
+
+*Default value*: ``40000000``
+
+*Recomended value*: ``60000000``
+
+*Description*:
+
+Internal heap end address used by the Simulator to manage its own memory. It is an adress in bytes.
```
Comment (frivieremicroej) [inline ApplicationDeveloperGuide/sectionSimulatorOptions.rst:144]: does it includes other memories declared above ?
I though there was a case where an option depends on another one

File: ApplicationDeveloperGuide/sectionSimulatorOptions.rst [hunk 11]
```diff
@@ -0,0 +1,151 @@
...
+
+..
```
Comment (frivieremicroej) [inline ApplicationDeveloperGuide/sectionSimulatorOptions.rst:146]: no options for immutable limit ?

File: ApplicationDeveloperGuide/sectionSimulatorOptions.rst [hunk 12]
```diff
@@ -0,0 +1,151 @@
...
+   | Copyright 2022, MicroEJ Corp. Content in this space is free
```
Comment (frivieremicroej) [inline ApplicationDeveloperGuide/sectionSimulatorOptions.rst:147]: no options for pin objects

**Review by frivieremicroej — CHANGES_REQUESTED** [review]

File: ApplicationDeveloperGuide/applicationOptions.rst (hunk 1 above)
Comment (bguedas) [reply ApplicationDeveloperGuide/applicationOptions.rst:111]: These options can be set as application option in properties files.

File: ApplicationDeveloperGuide/sectionSimulatorOptions.rst (hunk 2 above)
Comment (bguedas) [reply ApplicationDeveloperGuide/sectionSimulatorOptions.rst:32]: that's right

File: ApplicationDeveloperGuide/sectionSimulatorOptions.rst (hunk 5 above)
Comment (bguedas) [reply ApplicationDeveloperGuide/sectionSimulatorOptions.rst:51]: It is the system interned strings chars. It looks like it's only used by the debugger.

File: ApplicationDeveloperGuide/sectionSimulatorOptions.rst (hunk 6 above)
Comment (bguedas) [reply ApplicationDeveloperGuide/sectionSimulatorOptions.rst:66]: these are the app interned strings. I've modified the text a bit.

File: ApplicationDeveloperGuide/sectionSimulatorOptions.rst (hunk 8 above)
Comment (bguedas) [reply ApplicationDeveloperGuide/sectionSimulatorOptions.rst:112]: Yes we did, in JUnit tests for instance.

File: ApplicationDeveloperGuide/sectionSimulatorOptions.rst (hunk 12 above)
Comment (bguedas) [reply ApplicationDeveloperGuide/sectionSimulatorOptions.rst:147]: No, there is a default value which is hard coded, but the array grows automatically if needed.

File: ApplicationDeveloperGuide/sectionSimulatorOptions.rst (hunk 11 above)
Comment (bguedas) [reply ApplicationDeveloperGuide/sectionSimulatorOptions.rst:146]: No, they are stored within a hash map that grows automatically

File: ApplicationDeveloperGuide/sectionSimulatorOptions.rst (hunk 9 above)
Comment (bguedas) [reply ApplicationDeveloperGuide/sectionSimulatorOptions.rst]: I've removed this option description.

**Review by frivieremicroej — APPROVED** [review]

**Pull request merged**